common/middleware.py

```diff
@@ -10,29 +10,39 @@ def __init__(self, get_response=None):
     def __call__(self, request):
         response = self.get_response(request)
         try:
-            res = json.loads(response.content.decode())
+            content = json.loads(response.content.decode())
         except json.JSONDecodeError:
             return response
 
-        if request.method == 'GET':
-            # Use case for retrieval of non-exitent record
-            if 'data' in res and list(res['data'].values())[0] is None:
-                res = None
-            # Use case for retrieval of all records
-            elif 'data' in res and 'edges' in list(res['data'].values())[0]:
-                res = list(list(res['data'].values())[0].values())[0]
-                res = [list(item.values())[0] for item in res]
-            # Use case for retrieval of a single existing record
-            elif 'data' in res and isinstance(list(res['data'].values())[0], dict):
-                res = list(res['data'].values())[0]
-        elif request.method == 'POST':
-            # Use case for creating, deleting and updating a record
-            if 'data' in res and isinstance(list(list(res['data'].values())[0].values())[0], dict):
-                res = list(list(res['data'].values())[0].values())[0]
-            # Use case for deleting and updating a record
-            elif 'data' in res and list(list(res['data'].values())[0].values())[0] is None:
-                res = None
-
-        response.content = json.dumps(res)
+        if 'data' in content and request.method == 'GET':
+            flattened_content = {}
+            index = 0
+            for resource in list(content['data'].values()):
+                # Use case for retrieval of all records
+                if (resource and 'edges' in resource
+                        and isinstance(resource['edges'], list)
+                        and 'node' in resource['edges'][0]):
 
+                    flattened_content[list(content['data'].keys())[index]] = []
+                    for item in list(resource.values())[0]:
+                        flattened_content[
+                            list(content['data'].keys())[index]
+                        ].append(list(item.values())[0])
+
+                # Use case for retrieval a record
+                elif resource is None or isinstance(resource, dict):
+                    flattened_content[list(content['data'].keys())[index]] = resource
+
+                index += 1
```

Comment:
I also don't understand why we have to increment index when we are looping over a pre-defined set of values. Do we need this index variable somewhere?